src/awkward/highlevel.py

```diff
@@ -302,19 +302,22 @@ def __init__(
 
         elif isinstance(data, dict):
             fields = []
-            contents = []
+            _arrays = []
             length = None
             for k, v in data.items():
                 fields.append(k)
-                contents.append(Array(v).layout)
+                _arrays.append(Array(v))
                 if length is None:
-                    length = contents[-1].length
-                elif length != contents[-1].length:
+                    length = _arrays[-1].layout.length
+                elif length != _arrays[-1].layout.length:
                     raise ValueError(
                         "dict of arrays in ak.Array constructor must have arrays "
-                        f"of equal length ({length} vs {contents[-1].length})"
+                        f"of equal length ({length} vs {_arrays[-1].layout.length}). "
+                        "For automatic broadcasting use 'ak.zip' instead. "
                     )
-            layout = ak.contents.RecordArray(contents, fields)
+            layout = ak.contents.RecordArray([arr.layout for arr in _arrays], fields)
+            attrs = attrs_of(*_arrays, attrs=attrs)
```

Comment:
@pfackeldey - would this overwrite the `attrs` and `behavior` parameters of `__init__` if they are not `None`?

Reply:
Yes, but only if `_arrays` would have non-`None` `attrs` and `behavior` defined on them. This is what I mean with propagation of metadata. `ak.zip` does this aswell. I hope that makes sense?

Reply:
To be more precise: whatever is provided in the `__init__` for attrs and behavior will take precedence. In case they're `None` it will try to propagate them from the given input arrays. (And that's also how `ak.zip` works.)